File: romanToInt.js
/**
 * @param {string} s
 * @return {number}
 */
function romanToInt(s) {
  const romanValues = {
    M: 1000,
    D: 500,
    C: 100,
    L: 50,
    X: 10,
    V: 5,
    I: 1,
  };

  let result = 0;
  let len = s.length;

  for (let i = 0; i < len; i++) {
    let currentValue = romanValues[s[i]];
    let nextValue = romanValues[s[i + 1]];

    if (currentValue < nextValue) {
      result -= currentValue;
    } else {
      result += currentValue;
    }
  }

  return result;
}

const s = "MCMXCIV";
const result = romanToInt(s);
console.log(result);

File: romanToInt.c
#include<stdio.h>
#include<string.h>

int romanToInt(char* s) {
    int romanToValue(int roman){
        switch(roman){
            case 'M':
                return 1000;
            case 'D':
                return 500;
            case 'C':
                return 100;
            case 'L':
                return 50;
            case 'X':
                return 10;
            case 'V':
                return 5;
            case 'I':
                return 1;
            default:
                return 0;
        };
    };

    int result = 0;
    int len = strlen(s);

    for(int i=0; i<len; i++){
        int currentValue = romanToValue(s[i]);
        int nextValue = romanToValue(s[i+1]);

        if(currentValue<nextValue){
            result -= currentValue;
        }else{
            result +=currentValue;
        }
    }
    return result;
}

int main(){
    char* s = "MCMXCIV";
    int result = romanToInt(s);
    printf("%d", result);

    return 0;
}